racingcar/game.go

```diff
@@ -0,0 +1,98 @@
+package racingcar
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	minParticipants     = 1
+	maxParticipants     = 20
+	allowedMinTurnCount = 1
+	allowedMaxTurnCount = 100
+)
+
+type RaceCars []Car
+
+type Game struct {
+	Cars             RaceCars
+	maxTurnCount     int
+	currentTurnCount int
+}
```

Comment:
P2. 그리고 전반적으로 구현하신 내용을 보니까 `[]Car`보다는 `[]*Car` 타입으로 관리하시는게 더 적절해보입니다. NewCar, initCars, PlayTurn에서 포인터를 생성하고 dereference하는 과정을 줄일 수 있을 것 같아요.

P3. 참고로 타입명을 필드명으로 그대로 사용하는 것도 가능합니다. 익명필드(anonymous field)라고 불러요. [참고자료](https://bugoverdose.github.io/development/composition-in-go)

```suggestion
type Cars []*Car

type Game struct {
	Cars
	maxTurnCount     int
	currentTurnCount int
}
```

Reply:
헉 Car slice로 저장하다보니 뒤에서 인덱스로 다시 접근하고 문제였는데, 포인터 슬라이스로 할 생각을 못했었네요..!

anonymous field도 이렇게 가능한지 몰랐었는데, 감사합니다!!